lib/screens/home/widget/home_paging_widget.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:lydiaryanfluttersurvey/model/ui/survey_ui_model.dart';
+import 'package:lydiaryanfluttersurvey/screens/widgets/background_widget.dart';
+import 'package:smooth_page_indicator/smooth_page_indicator.dart';
+
+class HomePagingWidget extends StatefulWidget {
+  final List<SurveyUiModel> surveys;
+
+  const HomePagingWidget({super.key, required this.surveys});
+
+  @override
+  State<StatefulWidget> createState() => _HomePagingWidgetState();
+}
+
+class _HomePagingWidgetState extends State<HomePagingWidget> {
+  int _currentIndex = 0;
+
+  void _onPageChanged(int index) {
+    setState(() {
+      _currentIndex = index;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final PageController pageController = PageController();
+    return Stack(
+      children: [
+        PageView(
+          controller: pageController,
+          onPageChanged: _onPageChanged,
+          children: widget.surveys
+              .map((SurveyUiModel survey) => BackgroundWidget(
+                    image: Image.network("${survey.coverImageUrl}l").image,
+                  ))
+              .toList(),
+        ),
+        _buildSurveyItem(context, widget.surveys[_currentIndex], pageController)
+      ],
+    );
+  }
+
+  Widget _buildSurveyItem(
+    BuildContext context,
+    SurveyUiModel survey,
+    PageController controller,
+  ) {
+    return Stack(
+      children: [
+        SafeArea(
+          child: Padding(
+            padding: const EdgeInsets.all(20),
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                _buildSurveyHeader(context, survey),
+                const Spacer(),
+                _buildSurveyFooter(context, survey, controller),
+              ],
+            ),
+          ),
+        )
+      ],
+    );
+  }
+
+  Widget _buildSurveyHeader(BuildContext context, SurveyUiModel survey) {
+    return Row(
+      children: [
+        Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Text(
+              DateFormat("EEEE, MMMM dd").format(survey.activeAt).toUpperCase(),
+              style: Theme.of(context)
+                  .textTheme
+                  .titleLarge
+                  ?.copyWith(fontSize: 13),
+            ),
+            Text(
+              _daysSince(context, survey.activeAt),
+              style: Theme.of(context).textTheme.titleLarge,
+            )
+          ],
+          //TODO add user avatar
+        ),
+      ],
+    );
+  }
+
+  String _daysSince(BuildContext context, DateTime dateTime) {
```

Comment:
What do you think about creating an [extension](https://dart.dev/language/extension-methods) for this?